repo2docker/buildpacks/r.py

```diff
@@ -257,4 +257,15 @@ def get_assemble_scripts(self):
                 )
             ]
 
+        descriptionR_path = self.binder_path('DESCRIPTION')
+        if os.path.exists(descriptionR_path):
+            assemble_scripts += [
+                (
+                    "${NB_USER}",
+                    r"""
+                    R --quiet -e "devtools::install_git('.')"
```

Comment:
what if the `DESCRIPTION` is in a `binder/` folder? Should this be `R --quiet -e "devtools::install_git('{}'.format(descriptionR_path)`?

Also it seems there's an extra `"` in there at the end

Reply:
That's a good point, though it seems a bit unlike (would be like a `setup.py` hiding outside the main folder) but I've added that bit 👍

Reply:
We only act on the `setup.py` if it is in the top level and there is no `binder/`:

https://github.com/jupyter/repo2docker/blob/46f056a4e281bb84b0c6b6e321ad0f1879e89ddd/repo2docker/buildpacks/python/__init__.py#L69-L74

The discussion when we added it was that we want to enable the opportunistic "just try it out" of packages even if they don't know of binder (yet). However when the package maintainers are aware of binder and want to do something different the signal for that is to create `binder/` subdir to specify exactly how to setup the environment for this package and its examples.

I think we should take the same approach for `DESCRIPTION` files.

Reply:
That makes sense to me. In that case I'd also extend the `detect` method to run the buildpack if only the `DESCRIPTION` is available but no `runtime.txt`. In that case we'd need to decide which snapshot of MRAN to use though. Do you have any suggestions for that? I'd be tempted to just go for the latest one if nothing is specified?